inc/classes/class-settings.php

```diff
@@ -112,5 +122,31 @@ public function cse_id_field_cb() {
 		<input type="text" name="gcs_cse_id" value="<?php echo ! empty( $setting ) ? esc_attr( $setting ) : ''; ?>">
 		<?php
 	}
+
+	/**
+	 * Search Type field markup.
+	 *
+	 * @return void
+	 */
+	public function cse_search_type_field_cb() {
+
+		// Get the value of the setting we've registered with register_setting().
+		$setting = get_option( 'gcs_search_type' );
+
+		// Add slider toggle switch.
+		?>
+		<div class="switch-wrapper">
+			<span><?php esc_html_e( 'Custom Site Restricted Search API', 'search-with-google' ); ?></span>
+
+			<label class="switch">
+				<input id="search-type" type="checkbox" name="gcs_search_type" value="1" <?php checked( $setting, '1' ); ?>>
```

Comment:
@SH4LIN Are we sure the `$settings` will be the string here? Can we convert it into an integer and check here, also please QA this properly.

Reply:
@vishalkakadiya If we look into the function implementation whatever we pass through the checked function will be first converted to string through type casting and then it will be compared.

![image](https://github.com/rtCamp/search-with-google/assets/56588503/9883b9c1-5e2d-4703-9df1-09cf6afad6f2)

so whether or not it will be string or not it will eventually converted into string.

I have done the QA on this properly and it's working as expected.

Reply:
checked() function calls this _checked_selected_helper() function.
![image](https://github.com/rtCamp/search-with-google/assets/56588503/8e8dd344-a1a6-4f80-8e83-66dd94b7f345)

Reply:
@SH4LIN Cool, then pass `1` without an inverted comma, approving this PR.